File: python/web_tutorial/color_grid.py
from Bridges.ColorGrid import *
from Bridges.Bridges import *
from Bridges.Color import *
import sys

class color_grid:

    # This example illustrates using the Bridges color grid
    # We will build a checker grid using two different colors

    args = sys.argv[1:]

    # create the Bridges object, set credentials
    bridges = Bridges(args[1], args[2], args[3])

    if len(args) > 3:
        bridges.connector.set_server(args[4])

    # set title fo visualization
    bridges.setTitle("A Checkerboard Example Using Grid Type")

    width = 200
    height = 200

    blue = Color(col_name= "blue")
    red = Color(col_name= "red")

    cg = ColorGrid(width, height, red)

    # create a checkerboard pattern
    # 10x10 grid
    num_squares_x = 10
    num_squares_y = 10

    sq_width = width/num_squares_x
    sq_height = width/num_squares_y

    for j in range(num_squares_y):
        for k in range(num_squares_x):

            # use even/odd value of pixel to figure out the color of the squares
            x_even = k%2 == 0
            y_even = j%2 == 0

            if(y_even == True):
                if(x_even == True):
                    col = red
                else:
                    col = blue
            else:
                if(x_even == True):
                    col = blue
                else:
                    col = red

            # find the address of the square
            origin_x = k*sq_width
            origin_y = j*sq_height

            #color the square
            for y in range(int(origin_y), int(origin_y+sq_height)):
                for x in range(int(origin_x), int(origin_x+sq_width)):
                    cg.set(x, y, col)
                    x = x + 1
                y = y + 1

    bridges.set_data_structure(cg)

    bridges.visualize()

File: java/web_tutorial_mastercopy/color_grid.java
/**
 * Created by Kalpathi Subramanian
 * Date : 7/17/2018.
 *
 * krs@uncc.edu
 */
import java.lang.String;
import bridges.connect.Bridges;
import bridges.base.SLelement;
import bridges.base.Color;
import bridges.base.ColorGrid;

/**
 *  This example illustrates the use of the Grid type and its subclass, a ColorGrid.
 *  The example builds  a checkerboard with two different colors
 *
 */

public class color_grid {

	public static void main(String[] args) throws Exception {

		// This example illustrates using the Bridges color grid
		// We will build a checker grid using two different colors

		// initialize BRIDGES
#if TESTING

		Bridges bridges = new Bridges(Integer.parseInt(args[0]), args[1], args[2]);
		bridges.setServer(args[3]);

		// set title for visualization
		bridges.set_title("A CheckerBoard Example Using Grid Type");
		bridges.set_description("This is an example of the color grid with height and width 10. "
            +   "This example alternates between blue and red starting with red at the top left corner of the grid.");
			
#else
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID", 
										"YOUR_API_KEY");
#endif
		
		int width = 200, height = 200;

		Color blue = new Color("blue");
		Color red = new Color("red");

		ColorGrid cg  = new ColorGrid(width, height, red);

		// create a checkerboard pattern
		// 10 x 10 grid
		int num_squares_x = 10;
		int num_squares_y = 10;

		// compute square dimensions
		int sq_width = width / num_squares_x, sq_height = width / num_squares_y;

		for (int j = 0; j < num_squares_y;  j++)
			for (int k = 0; k < num_squares_x;  k++) {

				// use even/odd value of pixel to
				// figure out the color of the square
				Boolean x_even = (k % 2) == 0;
				Boolean y_even = (j % 2) == 0;

				Color col;

				if (y_even)
					col = (x_even) ? red : blue;
				else
					col = (x_even) ? blue : red;

				// find the address of the square
				int origin_x = k * sq_width;
				int origin_y = j * sq_height;

				// color the square
				for (int y = origin_y; y < origin_y + sq_height; y++)
					for (int x = origin_x; x < origin_x + sq_width; x++)
						cg.set(x, y, col);
			}

		bridges.setDataStructure(cg);

		bridges.visualize();
	}
}
